pages/deployment/[uuid].tsx

```diff
@@ -162,6 +165,10 @@ export default function Deployment() {
     await postEndpoint(`/api/v1/deployment/${deployment?.uuid}/reset-approvals`, {}).then((res) => res.json())
   }
 
+  const initialVersionRequested: ObjectId | VersionDoc = deployment.versions.filter(
+    (version: any) => version.version === deployment.metadata.highLevelDetails.initialVersionRequested
```

Comment:
I'm not happy using "any" here, but it seems TS is complaining if I try anything else. Might revisit this when I get time

Reply:
TS is complaining because `deployment.versions` is typed as `(Types.ObjectId | VersionDoc)[]`, so  you'd need to use a type guard in your filter to ensure you're working with `VersionDoc`s. 
```
// type guard
const isVersionDoc = (value: unknown): value is VersionDoc => value && value._id && value.version

// using type guard
// I've made the assumption that we only care about `VersionDoc`s as the UI should have the full object 
const initialVersionRequested = deployment.versions.filter(
  version => isVersionDoc(version) && version.version === deployment.metadata.highLevelDetails.initialVersionRequested
)[0]
```

Reply:
To add to the above, for completeness and ease of handling the optional `initialVersionRequested`, you could return a simple error page if `initialVersionRequested` is `undefined`. That way, for the rest of the JSX, you can work with it being a guaranteed value.